### PR TITLE
Pivotal #149412681: feature doc for `list_*` methods, add values-only option

### PR DESCRIPTION
**Pull request by swryan**
"Add feature doc/tests for list_outputs/list_inputs/list_residuals and add an option to the methods to print just the names of the variables"

Implemented additional functionality in the logger utility to be able to change output stream, etc.

Fixed an issue in `docutil` with embedding code that does not produce output

Some minor documentation clean-up

### CODE REVIEW

File: openmdao/docs/features/building_components/explicitcomp.rst [hunk 1]
```diff
@@ -16,6 +16,13 @@ ExplicitComponent Methods
 
 The implementation of each method will be illustrated using a simple explicit component that computes the output *area* as a function of inputs *length* and *width*.
 
+::
```
Comment (hwangjt) [inline openmdao/docs/features/building_components/explicitcomp.rst]: I was actually just thinking of doing this myself. By the way, isn't there an automatic way of doing this? Along the lines of:
```
.. autoclass:: openmdao.core.group.Group
    :noindex:
```

File: openmdao/core/system.py [hunk 2]
```diff
@@ -2454,24 +2423,27 @@ def list_outputs(self, explicit=True, implicit=True, out_stream=sys.stdout):
         implicit : bool, optional
             include outputs from implicit components. Default is True.
 
+        values : bool, optioanl
+            When True, display/return output values as well as names.
+
         out_stream : file_like
             Where to send human readable output. Default is sys.stdout.
```
Comment (swryan) [inline openmdao/core/system.py]: wilco

File: openmdao/docs/features/building_components/explicitcomp.rst (hunk 1 above)
Comment (swryan) [inline openmdao/docs/features/building_components/explicitcomp.rst]: that does import the code, but not in a box in the code style like the rest of the code.. may be possible to style it?

Comment (hwangjt) [inline openmdao/docs/features/building_components/explicitcomp.rst]: @kmarsteller , do you know if it's possible?

Comment (kmarsteller) [inline openmdao/docs/features/building_components/explicitcomp.rst]: are you talking about the stuff I have here: http://blue.readthedocs.io/en/latest/style_guide/doc_style_guide.html

Comment (hwangjt) [inline openmdao/docs/features/building_components/explicitcomp.rst]: I didn't mean to hold up the pull request for this. Let's just merge and clean up formatting later when we do a pass through all the docs

**Pull request merged**